picard/ui/mainwindow.py

```diff
@@ -944,6 +948,10 @@ def update_selection(self, objects=None):
                         }
                     self.set_statusbar_message(msg, mparms, echo=None,
                                                history=None)
+            elif isinstance(obj, Album):
+                obj.update_metadata_images()
```

Comment:
Does this really need to be called every time an album is selected? It really feels like this should be done after the images on the metadata objects are updated.